src/Microsoft.ApiDesignGuidelines.Analyzers/Core/EnumStorageShouldBeInt32.Fixer.cs

```diff
@@ -1,30 +1,64 @@
 // Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using System;
-using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.CodeActions;
+using System.Collections.Immutable;
 
 namespace Microsoft.ApiDesignGuidelines.Analyzers
 {
     /// <summary>
     /// CA1028: Enum Storage should be Int32
     /// </summary>
+
     public abstract class EnumStorageShouldBeInt32Fixer : CodeFixProvider
     {
+        protected abstract SyntaxNode GetTargetNode (SyntaxNode node);
+
         public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(EnumStorageShouldBeInt32Analyzer.RuleId);
 
         public sealed override FixAllProvider GetFixAllProvider()
         {
-            // See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
+            // Fixes all occurrences within within Document, Project, or Solution
             return WellKnownFixAllProviders.BatchFixer;
         }
 
         public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
         {
-            // This is to get rid of warning CS1998, please remove when implementing this analyzer
-            await new Task(() => { });
-            throw new NotImplementedException();
+            var title = MicrosoftApiDesignGuidelinesAnalyzersResources.EnumStorageShouldBeInt32Title;
+
+            // Get syntax root node
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            // No multiple overlapping diagnostics of this id. So get the only diagnostic.
+            var diagnostic = context.Diagnostics.Single();
+
+            // Register fixer
+            context.RegisterCodeFix(CodeAction.Create(title, c => ChangeEnumTypeToInt32Async(context.Document, diagnostic, root, c), equivalenceKey: title), context.Diagnostics.First());
         }
-    }
-}
\ No newline at end of file
+
+        private async Task<Document> ChangeEnumTypeToInt32Async(Document document, Diagnostic diagnostic, SyntaxNode root, CancellationToken cancellationToken)
+        {
+            var editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+            var generator = editor.Generator;
+
+            // Find syntax node that declares the enum
+            var diagnosticSpan = diagnostic.Location.SourceSpan;
+            var node = root.FindNode(diagnosticSpan);
+            var enumDeclareNode = editor.Generator.GetDeclaration(node, DeclarationKind.Enum);
```

Comment:
You already have the `generator` object in hand, so you can use it here, and below, instead of repeating `editor.Generator`.